include/LLVMSPIRVLib.h

```diff
@@ -106,6 +106,10 @@ std::unique_ptr<SPIRVModule> readSpirvModule(std::istream &IS,
                                              const SPIRV::TranslatorOpts &Opts,
                                              std::string &ErrMsg);
 
+/// \brief Returns the message associated with the error code.
+/// \returns empty string if no known error code is found.
+std::string getErrorMessage(int ErrCode);
```

Comment:
Do we want to have a separate API for this? or can we just modify getSpirvReport to return a error string instead of an Id?

Thanks

Reply:
We don't want getSpirvReport to return strings, it should be as lightweight as it's possible. So we need to keep integer error code in getSpirvReport, and I doubt even that it will be much used, I'd expect that in backend it would be ignored. So the main use for error codes and error messages is for human-readable info.

Re: testing -- Sure, after the merge of https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2298, we'll be able to add more tests. Until the merge of #2298 that's a draft PR.

Reply:
@asudarsa There is a test case now in test/negative/spirv_report_bad_input.spt for this API call